lib/poxa/web_hook/handler.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Poxa.WebHook.Handler do
+  use GenEvent
+  alias Poxa.WebHook.EventData
+  alias Poxa.WebHook.EventTable
+
+  @delay 1500
+
+  def init(_) do
+    EventTable.init
+    {:ok, []}
+  end
+
+  @doc false
+  def handle_event(%{event: :channel_vacated, channel: channel}, []) do
+    EventData.channel_vacated(channel) |> save_web_hook(@delay)
+  end
+
+  def handle_event(%{event: :channel_occupied, channel: channel}, []) do
+    EventData.channel_occupied(channel) |> save_web_hook
+  end
+
+  def handle_event(%{event: :member_added, channel: channel, user_id: user_id}, []) do
+    EventData.member_added(channel, user_id) |> save_web_hook
+  end
+
+  def handle_event(%{event: :member_removed, channel: channel, user_id: user_id}, []) do
+    EventData.member_removed(channel, user_id) |> save_web_hook(@delay)
+  end
+
+  def handle_event(%{event: :client_event_message, socket_id: socket_id, channels: channels, name: name, data: data}, []) do
+    for c <- channels do
+      EventData.client_event(c, name, data, socket_id, nil)
+    end |> save_web_hook
+  end
+
+  # Events that cannot trigger any webhook event
+  def handle_event(_, []), do: {:ok, []}
+
+  defp save_web_hook(event, delay) when not is_list(event), do: save_web_hook([event], delay)
+  defp save_web_hook(events, delay \\ 0) do
+    Enum.each events, fn(event) ->
```

Comment:
let's kill this and simply call EventTable.insert(event | events, delay) wherever you call save_web_hook. 

Can we also use nil as state as we have no need for a list?